From fac42bc36baad89d318b2c0e514cdcd3692434fd Mon Sep 17 00:00:00 2001
From: "Valery V. Vorotyntsev" <valery_vorotyntsev@xyratex.com>
Date: Fri, 20 Apr 2012 23:10:42 +0300
Subject: [PATCH 147/370] confc.c (request_fill): use c2_confc_ctx argument

The function assumes that the passed request is embedded in ctx anyway,
which would lead to errors should anybody try to call it with a
"stand-alone" request.

RB: r/714
---
 conf/confc.c | 7 +++----
 1 file changed, 3 insertions(+), 4 deletions(-)

diff --git a/conf/confc.c b/conf/confc.c
index aec1759..f075206 100644
--- a/conf/confc.c
+++ b/conf/confc.c
@@ -479,7 +479,7 @@ static int act(struct c2_confc_ctx *cx, const struct walk_ctx *wx,
 		C2_CNT_INC(wx->w_obj->co_nrefs);
 		break;
 	case C2_CS_MISSING:
-		request_fill(&cx->fc_req, wx->w_obj, wx->w_path);
+		request_fill(cx, wx->w_obj, wx->w_path);
 		if (wx->w_atdir) {
 			/* ensure list_status() did its job properly */
 			C2_ASSERT(XXX /* directory has 0 C2_CS_MISSING objects
@@ -728,11 +728,10 @@ static bool request_is_valid(const struct c2_conf_fetch *req)
 
 /** Fills c2_conf_fetch structure. */
 static void
-request_fill(struct c2_conf_fetch *req, const struct c2_conf_obj *origin,
+request_fill(struct c2_conf_ctx *ctx, const struct c2_conf_obj *origin,
 	     const struct c2_conf_pathcomp *path)
 {
-	struct c2_confc_ctx *ctx =
-		container_of(req, struct c2_confc_ctx, fc_req);
+	struct c2_conf_fetch *req = &ctx->fc_req;
 
 	C2_PRE(ctx_invariant(ctx));
 	C2_PRE(c2_conf_path_is_valid(origin, path, NULL));
-- 
1.8.3.2

